spec/lib/manageiq/providers/workflows/builtin_methods_spec.rb

```diff
@@ -17,14 +17,56 @@
     end
 
-    it "returns an error if _object_id isn't passed" do
-      runner_context = described_class.provision_execute(params, secrets, Floe::Workflow::Context.new({"Execution" => {"_object_type" => "MiqProvision"}}))
-      expect(runner_context).to include("running" => false, "success" => false, "output" => {"Error" => "States.TaskFailed", "Cause" => "Missing MiqRequestTask id"})
+    it "requires _object_id" do
+      runner_context = described_class.provision_execute(params, secrets, create_floe_context(:execution => {"_object_type" => "MiqProvision"}))
+      expect(runner_context).to include("running" => false, "success" => false, "output" => failed_task_status("Missing MiqRequestTask id"))
     end
+
+    it "requires a provisioning object" do
+      floe_context = create_floe_context(FactoryBot.create(:service_reconfigure_task, :request_type => "service_reconfigure"))
+      runner_context = described_class.provision_execute(params, secrets, floe_context)
+      expect(runner_context).to include("running" => false, "success" => false, "output" => failed_task_status(/Calling provision_execute on non-provisioning request/))
+    end
+
+    it "requires a provisioning object" do
+      task_options = {
+        :src_vm_id => FactoryBot.create(:vm_vmware, :ext_management_system => FactoryBot.create(:ems_vmware_with_authentication)).id
+      }
+      request = FactoryBot.create(:miq_provision_request, :with_approval)
+      request.miq_approvals.update_all(:state => "approved")
+      task = FactoryBot.create(:miq_provision_vmware, :clone_to_vm, :options => task_options, :miq_request => request)
+      floe_context = create_floe_context(task)
+      runner_context = described_class.provision_execute(params, secrets, floe_context)
+
+      expect(runner_context["miq_request_task_id"]).to eq(task.id)
+    end
+
+    it "updates task options" do
+      task_options = {
+        :src_vm_id => FactoryBot.create(:vm_vmware, :ext_management_system => FactoryBot.create(:ems_vmware_with_authentication)).id
+      }
+      request = FactoryBot.create(:miq_provision_request, :with_approval)
+      request.miq_approvals.update_all(:state => "approved")
+      task = FactoryBot.create(:miq_provision_vmware, :clone_to_vm, :options => task_options, :miq_request => request)
+      floe_context = create_floe_context(task, :input => {:param1 => 1, :param2 => 2})
+      described_class.provision_execute(params, secrets, floe_context)
+
+      expect(task.reload.options).to include(:param1 => 1, :param2 => 2)
+    end
+  end
+
+  def create_floe_context(object = nil, execution: nil, input: {})
+    execution ||= {"_object_id" => object&.id, "_object_type" => object&.class}.compact
+
+    Floe::Workflow::Context.new({"Execution" => execution}, :input => input).tap { |ctx| ctx.state["Input"] = input }
+  end
+
+  def failed_task_status(cause = nil, error: "States.TaskFailed")
```

Comment:
None of the callers actually pass error - is that intentional or was it missed?

Reply:
pretty much all errors will be `TaskFailed`.
So I was leaving this open, but probably not necessary.

You want me to simplify?

Reply:
No, it's fine - I just wasn't sure if it was intentional or an oversight.

Reply:
holding off for now - may just drop in the future when adding more specs if it is never used